src/strings/needleman_wunsch/helpers.rs

```diff
@@ -0,0 +1,313 @@
+//! Helper functions for the Needleman-Wunsch algorithm.
+
+use crate::{number::UInt, strings::Penalties};
+
+/// The direction of best alignment at a given position in the DP table
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Dir {
+    /// Diagonal (Up and Left) for a match.
+    D,
+    /// Up for a gap in the first sequence.
+    U,
+    /// Left for a gap in the second sequence.
+    L,
+}
+
+/// The type of edit needed to turn one sequence into another.
+pub enum Edit {
+    /// Delete a character at the given index.
+    Del(usize),
+    /// Insert a character at the given index.
+    Ins(usize, char),
+    /// Substitute a character at the given index.
+    Sub(usize, char),
+}
+
+/// Computes the Needleman-Wunsch dynamic programming table for two sequences.
```

Comment:
This may be overly nitpicky, but since this is a public function, should we acknowledge in our documentation that the table is "diagonal prioritizing" and that the same penalties can yield different tables?

Reply:
The `helpers` module is private so this does not matter.